service/storage_service.cc

```diff
@@ -922,7 +947,9 @@ future<> storage_service::handle_state_normal(inet_address endpoint) {
                 do_remove_node(endpoint);
             }
         } else if (existing && *existing == endpoint) {
+            slogger.info("Existing node and replacing node is the same {}", endpoint);
             tmptr->del_replacing_endpoint(endpoint);
+            tmptr->update_host_id(host_id, endpoint);
```

Comment:
How come this is needed?

Reply:
Try replacing the node using the same ip address. Without this, we will see the new node will use the old host_id.